### PR TITLE
Inline legacyColumnMapping method in Redshift

### PR DESCRIPTION
**Pull request by ebyhr**
Unfortunately, there is no tests in Redshift connector.

### CODE REVIEW

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java [hunk 1]
```diff
@@ -77,8 +97,61 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
...
+                }
+                return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
+
+            case Types.CHAR:
+                return Optional.of(defaultCharColumnMapping(typeHandle.getRequiredColumnSize(), false));
+
+            case Types.VARCHAR:
+                return Optional.of(defaultVarcharColumnMapping(typeHandle.getRequiredColumnSize(), false));
+
+            case Types.BINARY:
```
Comment (hashhar) [inline plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:138]: Redshift docs don't show binary columns as being supported.

**Review by hashhar — COMMENTED** [review]
Would it be safer to inline the method using IDE? It would add some dead code but we can prevent regressions. And we can clean up the extra code after local tests?

@findepi WDYT?

**Comment by findepi** [comment]
> Would it be safer to inline the method using IDE?

I would actually move `legacyColumnMapping` into Redshift (adorned with some comment), so that the code doesn't look as-if explicit mapping were implemented.

**Comment by ebyhr** [comment]
@hashhar @findepi Just inlined with TODO comment.

**Comment by ebyhr** [comment]
CI #8432

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java (hunk 1 above)
Comment (hashhar) [reply plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:150]: Since we are inlining - this comment can be ignored.

**Review by hashhar — APPROVED** [review]
Looks good to me.

Or did @findepi want to copy the method (instead of inline) into RedshiftClient?

**Comment by findepi** [comment]
> Or did @findepi want to copy the method (instead of inline) into RedshiftClient?

i'd prefer copy without inlining, otherwise it's unclear why we have  eg all these options. which ones are actually applicable to redshift?

``
            case Types.VARCHAR:
            case Types.NVARCHAR:
            case Types.LONGVARCHAR:
            case Types.LONGNVARCHAR:
```


when you keep legacy mapping as separate method (with a comment that's it's not redshift-specific, just copied), it's easier to grasp the intention

**Comment by ebyhr** [comment]
@findepi I pushed another commit to remove 2 methods from base JDBC module. Could you take a look when you have time?

**Review by findepi — APPROVED** [review]

**Pull request merged**